src/en/weebdex/src/eu/kanade/tachiyomi/extension/en/weebdex/WeebDex.kt

```diff
@@ -0,0 +1,287 @@
+package eu.kanade.tachiyomi.extension.en.weebdex
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import okhttp3.Request
+import okhttp3.Response
+import org.json.JSONArray
+import org.json.JSONObject
+import java.text.SimpleDateFormat
+import java.util.Locale
+import java.util.TimeZone
+
+class WeebDex : HttpSource() {
+    override val name = "WeebDex"
+    override val baseUrl = "https://api.weebdex.org"
+    override val lang = "en"
+    override val supportsLatest = true
+
+    // Rate limit by WeebDex API is 5reqs/sec. Its toned down to 3 here just in case.
+    override val client = network.cloudflareClient.newBuilder()
+        .rateLimit(3)
+        .build()
+
+    // -------------------- Popular --------------------
+
+    override fun popularMangaRequest(page: Int): Request {
+        val url = "$baseUrl/manga".toHttpUrlOrNull()!!.newBuilder()
+            .addQueryParameter("page", page.toString())
+            .addQueryParameter("sort", "views")
+            .addQueryParameter("order", "desc")
+            .addQueryParameter("hasChapters", "1")
+            .build()
+        return GET(url.toString())
```

Comment:
Send headers, same for the other requests

```suggestion
        return GET(url.toString(), headers)
```

Reply:
Ok, i am passing headers, a bit similar to mangadex, but i didnt know how much to pass or how much was necessary. Stuff seems to work just fine for now with just this, so im keeping this for now:
```
  override fun headersBuilder(): Headers.Builder = Headers.Builder()
        .add("User-Agent", "Tachiyomi " + System.getProperty("http.agent"))
        .add("Referer", "$baseUrl/")
```

If we end up needing more in the future, we'll add them i guess?